foo/lib/config/configloader.py

```diff
@@ -0,0 +1,98 @@
+import os
+from typing import Optional
+
+import boto3
+import yaml
+
+
+class ConfigLoaderException(Exception):
+    pass
+
+
+class Config:
+
+    def __init__(self, config: dict, source: str):
+        self._config = config
+        self._source = source
+
+    @property
+    def source(self):
+        return self._source
+
+    def __repr__(self):
+        return (f'Config(config={self._config!r}'
+                f'       source={self.source!r})')
+
+    def __getattr__(self, name):
+        return self._config[name]
+
+    def __getitem__(self, name):
+        return self._config[name]
+
+    def get(self, name, default=None):
+        return self._config.get(name, default)
+
+
+class ConfigLoader:
+
+    """Loads configuration files.
+
+    Source of the config depends on the currently set environment (either
+    'test' or 'prod', as dictated by the *env* file.
+    """
+
+    S3_BUCKET = 'configs'
+    _instance = None
+    _instance_ready = False
+
+    # Make this class a singleton since usually config is loaded only once.
```

Comment:
Have you considered making the config singleton and writing the loader as plain functions? 
Suggested because we have "local" and "s3" in one class which seems a bit odd but its no biggie